be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -145,6 +145,9 @@ class SinkBuffer {
 
     void decrease_running_sinkers() {
         if (--_num_remaining_sinkers == 0) {
+            // _is_finishing is used to return early for _try_to_trigger_rpc_task, RPC task _process, and _send_rpc.
```

Comment:
I guess maybe it's unnecessary. because

1. `_num_remaining_sinkers` is atomic, which mean `(--_num_remaining_sinkers) == 0` will be triggered only once.
2. and at that one time, if `is_finishiing` is atomic value, then read/write is also atomic.

I would suggest to keep `std::atomic<bool> _is_finishing = false;` and 

```
void decrease_running_sinkers() {
        if (--_num_remaining_sinkers == 0) {
             _is_finishing = true;
        } 
}
```

```

Reply:
_is_finishing must under lock for some guarantees, you can search the file with key word 「guarantee」

Reply:
> 
_is_finishing must under lock for some guarantees, you can search the file with key word 「guarantee」

Reply:
But in this case, `_is_finishing` does not need to be protected by lock, it's atomic.